solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts

```diff
@@ -4,111 +4,154 @@
+  const paabHelper = new PaabHelper();
   let adminSession: ClientSession;
+  let pa1Session: ClientSession;
+  let rs1Session: ClientSession;
+  let project1Id: string;
+  let project2Id: string;
   let sshKeyId: string;
-  let project: { id: string };
-  const randomTextGenerator = new RandomTextGenerator(setup.getSettings().get('runId'));
 
-  beforeEach(async () => {
+  beforeAll(async () => {
+    ({ adminSession, pa1Session, rs1Session, project1Id, project2Id } = await paabHelper.createResources());
     expect.hasAssertions();
-    adminSession = await setup.getDefaultAdminSession();
-    const { data: costCenter } = await adminSession.resources.costCenters.create({
-      name: randomTextGenerator.getFakeText('fakeCostCenterName'),
-      accountId: setup.getSettings().get('defaultHostingAccountId'),
-      description: 'a test object'
-    });
-
-    const { data } = await adminSession.resources.projects.create({
-      name: randomTextGenerator.getFakeText('fakeProjectName'),
-      description: 'Project for list users for project tests',
-      costCenterId: costCenter.id
-    });
-
-    project = data;
   });
 
-  afterEach(async () => {
-    await setup.cleanup();
+  afterAll(async () => {
+    await paabHelper.cleanup();
   });
 
-  // TODO: multiple user session support has to exist for this test
-  describe('when current user does not own key', () => {
-    let existingSshKeyId: string;
-    let secondaryUserId: string;
-
-    beforeEach(async () => {
-      // This needs to happen in not the defaultAdminSession so the default admin does not own this key
-      // This user session must still have access to the project used
-      // const {data: existingSshKey} = await adminSession.resources.projects.project(project.id).sshKeys().create();
-      // secondaryUserId = adminSession.getUserId()!;
-      // existingSshKeyId = existingSshKey.sshKeyId;
+  describe('when key does not exist', () => {
+    let invalidSshKeyId: string;
+    const testBundle = [
+      {
+        username: 'projectAdmin1',
+        session: () => pa1Session,
+        projectId: () => project1Id
+      },
+      {
+        username: 'researcher1',
+        session: () => rs1Session,
+        projectId: () => project1Id
+      }
+    ];
+
+    beforeEach(() => {
+      invalidSshKeyId = `sshkey-0000000000000000000000000000000000000000000000000000000000000000`;
     });
 
-    test.skip('it throws 403 error', async () => {
+    test.each(testBundle)('within a valid project', async (testCase) => {
+      const { username, session: sessionFunc, projectId: projectIdFunc } = testCase;
+      const session = sessionFunc();
+      const projectId = projectIdFunc();
+
+      console.log(`as ${username}`);
+
       try {
-        await adminSession.resources.projects.project(project.id).sshKeys().sshKey(existingSshKeyId).purge();
+        await session.resources.projects.project(projectId).sshKeys().sshKey(invalidSshKeyId).delete();
```

Comment:
I think this should be a purge

Reply:
I think Kevin just merged that code, thats why in the conflict the purge was there

Reply:
looks like the latest have both delete and purge? Looks like the one corresponding to sshKey delete is delete.

Reply:
switched to purge